kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ZooKeeperClientProvider.scala

```diff
@@ -55,6 +59,35 @@ object ZooKeeperClientProvider extends Logging {
       case UNTIL_ELAPSED => new RetryUntilElapsed(maxSleepTime, baseSleepTime)
       case _ => new ExponentialBackoffRetry(baseSleepTime, maxRetries)
     }
+    val chrootIndex = connectionStr.indexOf("/")
+    val chrootOption = {
+      if (chrootIndex > 0) Some(connectionStr.substring(chrootIndex))
+      else None
+    }
+    // make sure zookeeper chroot path exists
+    chrootOption.foreach { chroot =>
+      val zkConnectionForChrootCreation = connectionStr.substring(0, chrootIndex)
+      val clonedConf = KyuubiConf(false)
+      clonedConf.set(HA_ZK_QUORUM, zkConnectionForChrootCreation)
```

Comment:
We can remove this logic and add a parameter `createChrootIfNecessary: Boolean = false` to the method to avoid checking chroot everytime.

Reply:
Nice advice, I will refine this if the other reviewers accept.